proposals/0082-relax-transaction-constraints.md

```diff
@@ -0,0 +1,173 @@
+of blocks; if any constraints are broken, block-validators will mark the block
+as invalid.
+Many of these constraints are necessary, but some of them are not, and lead to
+additional complexity in the protocol.
+This proposal aims to relax some of the constraints at the individual
+transaction level, in order to simplify the protocol, and give more flexibility
+to block-producer and block-validator implementations.
+
+More specifically, this proposal aims to relax the constraints which require
+account state in order to determine the validity of a transaction's inclusion.
+The reason these constraints are targetted specifically, is that reliance on
+account state necessarily requires synchronous execution within the protocol.
+This proposal on its' own, will not enable asynchronous execution, but it will
+remove one of the barriers to asynchronous execution.
+
+## Alternatives Considered
+
+1. Do nothing
+    - This is the simplest option, as we could leave the protocol as is.
+    However, this leaves the protocol more complex than it needs to be.
+2. Relax all but fee-paying constraint
+    - This was actually the initial proposal, but it was decided that it would
+    be better to relax as many constraints as possible, rather than just some.
+    Any reliance on account state, means the protocol requires synchronous
+    execution.
+3. Additionally, relax the address lookup table resolution constraint
+    - This was considered, since it is a transaction-level constraint that is
+    depdendent on account-state. However, due to entry-level and block-level
+    constraints that rely on the address lookup table resolution, this
+    constraint cannot easily be relaxed without also relaxing those
+    constraints.
+
+## New Terminology
+
+None
+
+## Detailed Design
+
+Specifically, this proposal relaxes constraints that a transaction included in
+a block must:
+
+1. Have a fee-payer with enough funds to pay fees
+2. Have a valid nonce account, if specified
+3. Have program accounts that:
+   1. exist
+   2. are executable
+4. Have writable accounts that are:
+   1. not executable, unless owned by
+   `BPFLoaderUpgradeab1e11111111111111111111111`
+   2. if owned by `BPFLoaderUpgradeab1e11111111111111111111111`,
+   then `BPFLoaderUpgradeab1e11111111111111111111111` must be included
+   3. if owned by `Stake11111111111111111111111111111111111111`, then the
+   current slot must not be within the epoch stakes reward distribution period
+5. Have executable `BPFLoaderUpgradeab1e11111111111111111111111` owned accounts
+with `UpgradeableLoaderState::Program` state, and derived program data account
+that exist
+6. Have a call chain depth of 5 or less
+7. Have no builtin loader ownership chains
+(pending `4UDcAfQ6EcA6bdcadkeHpkarkhZGJ7Bpq7wTAiRMjkoi`)
+8. Have total loaded data size which does not exceed
+requested_loaded_accounts_data_size_limit
+(pending `DdLwVYuvDz26JohmgSbA7mjpJFgX5zP2dkp8qsF2C33V`)
```

Comment:
Tried to be as specific as possible in the document itself, but these constraints (3-8) can be summarized as "transaction must be executable"